suse_migration_services/prechecks/kernels.py

```diff
@@ -54,24 +56,50 @@ def multiversion_and_multiple_kernels():
                         'Please ensure it is set as:\n'
                         "'multiversion.kernels = latest,running'")
         elif set(kernel_multi_version_settings.split(',')) != {'running', 'latest'}:
-            log.warning('The config option multiversion.kernels is not set '
-                        'correctly in /etc/zypp/zypp.conf. It is currrently '
-                        'set as \n%s.\nPlease ensure it is set as:\n'
-                        "'multiversion.kernels = latest,running'",
-                        kernel_multi_version_settings)
+            log.warning("The config option multiversion.kernels is not set "
+                        "correctly in /etc/zypp/zypp.conf. It is currrently "
+                        "set as:\n'multiversion.kernels = %s'.", kernel_multi_version_settings)
+            if fix:
+                log.warning("The '--fix' option was provided, "
+                            "setting 'multiversion.kernels = latest,running' ")
+                existing = 'multiversion.kernels = ' + kernel_multi_version_settings
+                correction = 'multiversion.kernels = latest,running'
+                sed_arg = 's/' + existing + '/' + correction + '/'
+                try:
+                    Command.run(["sed", "-i", sed_arg, Defaults.get_zypp_config_path()])
+                except DistMigrationCommandException:
+                    log.error("ERROR: Unable to update /etc/zypp/zypp.conf")
+
+            else:
+                log.warning("Please ensure it is set as:\n"
+                            "'multiversion.kernels = latest,running'")
 
     kernel_type = 'kernel-default'
-    running_kernel = Command.run(["uname", "-r"])
+    running_kernel = Command.run(["uname", "-r"]).output
 
-    if 'azure' in running_kernel.output:
+    if 'azure' in running_kernel:
         kernel_type = 'kernel-azure'
 
-    installed_kernels = Command.run(["rpm", "-qa", kernel_type])
-    multiple_kernels = len(installed_kernels.output.splitlines()) > 1
+    installed_kernels = Command.run(["rpm", "-qa", kernel_type]).output.splitlines()
+    multiple_kernels = len(installed_kernels) > 1
 
     if multiple_kernels:
-        log.warning(
-            'Multiple kernels have been detected on the system:\n'
-            '%s\n'
-            'Please remove all kernels other than the currrent running kernel, '
-            '%s\n', installed_kernels.output, running_kernel.output)
+        log.warning('Multiple kernels have been detected on the system:\n'
+                    '%s', '\n'.join(installed_kernels))
+        if fix:
+            running_kernel_version = kernel_type + '-' + \
+                re.search(r'([0-9.-]*(?=[-][a-zA-Z]))', running_kernel).group()
+
+            log.warning("The '--fix' option was provided, "
```

Comment:
This is `info` the user told us to do something and we are giving feedback

Reply:
I have updated this, please review